src/Cardano/BM/Trace.lhs

```diff
@@ -233,14 +229,15 @@ traceConditionally
     -> LogObject
     -> m ()
 traceConditionally logTrace@(ctx, _) msg@(LogObject _ (LogMessage item)) = do
-#ifdef MemoizeSeverity
-    minsev <-liftIO $ getSeverity (configuration ctx) (minSeverity ctx) (loggerName ctx)
-#else
-    globminsev <- liftIO $ Config.minSeverity (configuration ctx)
-    globnamesev <- liftIO $ Config.inspectSeverity (configuration ctx) (loggerName ctx)
-    let minsev =
-            max (minSeverity ctx) $ max globminsev $ fromMaybe Debug globnamesev
-#endif
+    let conf = configuration ctx
+    memoizeSeverity <- liftIO $ getMemoizeSeverity conf
+    minsev <- liftIO $ if memoizeSeverity
```

Comment:
this changes behaviour! `getSeverity` will test all paths up to the root, while the other direct functions will only evalute for the given logger name.